aiohttp/client_reqrep.py

```diff
@@ -340,6 +340,9 @@ def update_headers(self, headers: Optional[LooseHeaders]) -> None:
         netloc = cast(str, self.url.raw_host)
         if helpers.is_ipv6_address(netloc):
             netloc = f"[{netloc}]"
+        elif netloc.endswith("."):
+            # Strip trailing dot. See https://github.com/aio-libs/aiohttp/issues/3636.
+            netloc = netloc[:-1]
```

Comment:
This needs coverage, but also it'd make sense to apply `str.rstrip()` like urllib3 does, to handle multiple trailing dots like Google Chrome.